rust/benches/revcomp_benchmark.rs

```diff
@@ -0,0 +1,37 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use revcomp::*;
+use std::hint::black_box;
+
+pub fn revcomp_array_unsafe(fwd: &[u8]) -> Vec<u8> {
+    let length = fwd.len();
+    let mut dest = vec![0u8; length];
+
+    unsafe {
+        for index in 0..length {
+            *dest.get_unchecked_mut(length - index - 1) =
+                *revcomp::COMPLEMENT.get_unchecked(*fwd.get_unchecked(index) as usize);
+        }
+    }
+
+    return dest;
+}
+
+fn bench_variants(c: &mut Criterion) {
+    let mut group = c.benchmark_group("revcomp");
+    group.bench_function("iterator", |b| {
+        b.iter(|| revcomp_iter(black_box("abcdghkmnrstuvwyABCDGHKMNRSTUVWY".as_bytes())))
```

Comment:
I believe we should benchmark on longer strings, at least 150bp.

Reply:
Actually, we do not need the benchmark at all. I admit that it was a bit for my own education. I expected a "c-variant" to be much faster than the iterator variant. This is also why I started with a more or less direct translation of your code. But the compiler is probably too good at optimizing :D 

I am happy to adopt to only have the iterator variant